aiida/work/launch.py

```diff
@@ -14,42 +14,58 @@
 
 __all__ = ['run', 'run_get_pid', 'run_get_node', 'submit']
 
-_persister = None
-
 
 def submit(process_class, **inputs):
-    assert not utils.is_workfunction(process_class), "Cannot submit a workfunction"
+    """
+    Submit the process with the supplied inputs to the daemon runner immediately returning control to
+    the interpreter. The return value will be the calculation node of the submitted process
 
-    # Use context manager to make sure connection is closed at end
+    :param process: the process class to submit
+    :param inputs: the inputs to be passed to the process
+    :return: the calculation node of the process
+    """
+    assert not utils.is_workfunction(process_class), 'Cannot submit a workfunction'
+
+    # Use a context manager to make sure connection is closed at end
     with rmq.new_blocking_control_panel() as control_panel:
```

Comment:
Shouldn't that also use the runner implementation? I think right now builders are not handled here.

Reply:
Yup, that won't work. I just reworked all the logic of ensuring the process class and inputs in #1404 which is blocked, because it depends on this PR. Ideally, I would rewrite `submit` to also go through the `Runner.submit` because then all that logic is nicely isolated. I have to check if I can get rid of the control panel way it used now though

Reply:
Ok, since that was already broken before I guess this is good to merge.

Reply:
It appears I can't approve since it's out of date...